plugins/events-backend-module-http/src/service/HttpEndpointEventPublisher.ts

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2022 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { errorHandler } from '@backstage/backend-common';
+import { Config } from '@backstage/config';
+import { EventPublisher, EventBus } from '@backstage/plugin-events-node';
+import express from 'express';
+import Router from 'express-promise-router';
+import { Logger } from 'winston';
+
+/**
+ * Publishes events received from their origin (e.g., webhook events from an SCM system)
+ * via HTTP endpoint and passes the request body as event payload to the registered subscribers.
+ *
+ * @public
+ */
+export class HttpEndpointEventPublisher implements EventPublisher {
+  private eventBus?: EventBus;
+
+  static fromConfig(env: {
+    config: Config;
+    logger: Logger;
+  }): HttpEndpointEventPublisher {
+    const topics = env.config.getStringArray('events.publishers.http.topics');
+    return new HttpEndpointEventPublisher(env.logger, topics);
+  }
+
+  private constructor(private logger: Logger, private topics: string[]) {}
+
+  async connect(eventBus: EventBus): Promise<void> {
+    this.eventBus = eventBus;
+  }
+
+  createRouter(): express.Router {
+    const router = Router();
+    router.use(express.json());
+    router.use(errorHandler());
+
+    this.topics.forEach(topic => this.addRouteForTopic(router, topic));
+
+    return router;
+  }
+
+  private addRouteForTopic(router: express.Router, topic: string): void {
+    const path = `/${topic}`;
```

Comment:
Drive-by review: thinking that we're hogging the routes a bit much here right? Perhaps we could register it under something like `/_http/v1/topics/${topic}`? Where `_http` is derived from the module ID. Another way to do that could also be to expose an extension point like `eventsRouterExtensionPoint` from the events backend, with `eventsRouter.addRouter('http', router)`, but for this use-case I'm feeling that's a bit overkill.

Reply:
I've added `/http` so that full endpoints will look like `/api/events/http/{topic}`.
Adding the moduleId / name seemed reasonable and sensible - in case of other future modules want to do add routes, too.

Reply:
The thing then, is if the `events` backend wants to add routes of its own, that collide with the module names. What if you want an `/api/events/health` for example. So `/api/events/module/http/{topic}` isn't crazy I'd say

Reply:
ok, sure. I can add `/module` as well :)